### DISCUSSION TITLE
How to cancel job `A` if job `B` is running

### DISCUSSION THREAD

**Discussion by podviaznikov** [dagster-io/dagster#18489] [Q&A] [answered]
I have this code for DBT

```
@dbt_assets(manifest=DBT_MANIFEST_PATH)
def wave_snowflake_dbt_assets(
    context: AssetExecutionContext, dbt: DbtCliResource, config: WaveDbtConfig
):

    if config.skip_when_job_is_running is not None:
        run_records = context.instance.get_run_records(
            RunsFilter(
                job_name=config.skip_when_job_is_running,
                statuses=[DagsterRunStatus.STARTED],
            )
        )

        print("current run record", len(run_records))
        if len(run_records) > 0:
            # TODO I think we need to set status to canceled ot skipped
            return
...            
 ```
 
 What I want is to cancel schedule run of one job when another job is running.
 
 Currently at the beginning I'm checking if any of the jobs are running and do "return".
 
 This works but it still reports job as succesfull but with 0 assets. Ideally I would want to to set status of this job to skipped or canceled. Is this possible?

**Comment by rexledesma** [comment, marked as answer]
Take a look at our guide for limiting concurrency in your Dagster data pipelines: https://docs.dagster.io/guides/limiting-concurrency-in-data-pipelines.

In your case, you can just limit the concurrency of your job to `1` to ensure that only one job is running at any given time.

**Reply by podviaznikov** [reply]
that is not what I want. I want to skip job A if job B is running.

**Reply by rexledesma** [reply]
Got it. It wasn't clear that you were talking about two jobs, job A and job B.

If you want to explicitly cancel the run, then you can send an API call to the Dagster server to cancel the run, using GraphQL: https://docs.dagster.io/concepts/webserver/graphql#terminate-an-in-progress-run.

You can get the existing run id through `context.run_id`.

**Reply by the4thamigo-uk** [reply]
> In your case, you can just limit the concurrency of your job to 1 to ensure that only one job is running at any given time.

@rexledesma I wonder if you could you explain exactly how this is done as I cant seem to get this working, and this is exactly what I want to do?

(To be clear, I read the above to mean that, _at most only one run of a given job can execute concurrently_)

**Reply by rexledesma** [reply]
I linked the instructions above: https://docs.dagster.io/guides/limiting-concurrency-in-data-pipelines#limiting-specific-runs-using-tags.

What is not working for you? Can you confirm that your Dagster daemon and Dagster webserver have the proper configuration in `dagster.yaml`? And that your runs are tagged with the key you've set in configuration?

**Reply by the4thamigo-uk** [reply]
Ah ok, I think I was just very confused here (it's probably just me but I found the docs quite confusing, and it wasn't clear to me that this was even possible without trying the experimental feature at the bottom of the page), so good to see that you can confirm this should work. I think I was expecting tag concurrency controls to apply to asset materializations when I trigger them from the UI. Also I was expecting backfill runs issued from the UI to also apply the concurrency rules, but it seems they are treated differently? When I issue run requests from my sensor I can see them queued up now in the _Runs_ view in the UI.

**Reply by the4thamigo-uk** [reply]
Thank you so much for replying, it gave me some confidence that it _should_ be working, so I could reassess my assumptions.
